pkg/utils/nodeutils_test.go

```diff
@@ -227,4 +227,29 @@ var _ = Describe("Nodeutils", func() {
 			Entry("@all", "@all", map[string]string{}),
 		)
 	})
+
+	Context("MachineConfig Pool with no node selector", func() {
+		targetNodeSelector := map[string]string{
+			"test-node-role": "",
+		}
+
+		mcp := &mcfgv1.MachineConfigPool{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "MachineConfigPool",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "invalidpool",
+			},
+			Spec: mcfgv1.MachineConfigPoolSpec{
+				Paused: false,
+			},
+		}
+		It("Get correct custom KC name", func() {
+			isMatchingPool := utils.McfgPoolLabelMatches(targetNodeSelector, mcp)
+			Expect(isMatchingPool).To(BeFalse())
```

Comment:
This evaluates to `false` because the `MachineConfigPool` we're defining on line 236 doesn't have a `NodeSelector` key, correct?

Do we also want to have a case where we ensure there is a `NodeSelector` key but `MatchLabels` is `nil`?